src/lib/voiceSearchHelper/__tests__/index-test.ts

```diff
@@ -189,6 +192,6 @@ describe('VoiceSearchHelper', () => {
     });
     voiceSearchHelper.toggleListening();
     voiceSearchHelper.dispose();
-    expect(voiceSearchHelper.isListening()).toBe(false);
```

Comment:
What was wrong with this assertion? IMO it was better because we weren't relying on implementation detail.

Reply:
`isListening` relies on `voiceSearchHelper.getState().status`.
Since we're not updating the `state` of `voiceSearchHelper` on `dispose`, the `status` will remain as `recognizing` which results `isListening` returning `true`. And that way we cannot test this. So I had to check if `stop` has been called.